cannon/mipsevm/program/patch.go

```diff
@@ -73,16 +68,26 @@ func PatchStack(st mipsevm.FPVMState) error {
 	}
 
 	// init argc, argv, aux on stack
-	storeMem(sp+4*1, 0x42)   // argc = 0 (argument count)
-	storeMem(sp+4*2, 0x35)   // argv[n] = 0 (terminating argv)
-	storeMem(sp+4*3, 0)      // envp[term] = 0 (no env vars)
-	storeMem(sp+4*4, 6)      // auxv[0] = _AT_PAGESZ = 6 (key)
-	storeMem(sp+4*5, 4096)   // auxv[1] = page size of 4 KiB (value) - (== minPhysPageSize)
-	storeMem(sp+4*6, 25)     // auxv[2] = AT_RANDOM
-	storeMem(sp+4*7, sp+4*9) // auxv[3] = address of 16 bytes containing random value
-	storeMem(sp+4*8, 0)      // auxv[term] = 0
+	storeMem(sp+4*0, 1)       // argc = 1 (argument count)
+	storeMem(sp+4*1, sp+4*21) // argv[0]
+	storeMem(sp+4*2, 0)       // argv[1] = terminating
+	storeMem(sp+4*3, sp+4*14) // envp[0] = x (offset to first env var)
+	storeMem(sp+4*4, 0)       // envp[1] = terminating
+	storeMem(sp+4*5, 6)       // auxv[0] = _AT_PAGESZ = 6 (key)
+	storeMem(sp+4*6, 4096)    // auxv[1] = page size of 4 KiB (value) - (== minPhysPageSize)
+	storeMem(sp+4*7, 25)      // auxv[2] = AT_RANDOM
+	storeMem(sp+4*8, sp+4*10) // auxv[3] = address of 16 bytes containing random value
+	storeMem(sp+4*9, 0)       // auxv[term] = 0
+
+	_ = st.GetMemory().SetMemoryRange(sp+4*10, bytes.NewReader([]byte("4;byfairdiceroll"))) // 16 bytes of "randomness"
+
+	// append 3 extra zero bytes to end at 4-byte alignment
+	envar := append([]byte("GODEBUG=memprofilerate=0"), 0x0, 0x0, 0x0, 0x0)
+	_ = st.GetMemory().SetMemoryRange(sp+4*14, bytes.NewReader(envar))
 
-	_ = st.GetMemory().SetMemoryRange(sp+4*9, bytes.NewReader([]byte("4;byfairdiceroll"))) // 16 bytes of "randomness"
+	// 24 bytes for GODEBUG=memprofilerate=0 + 4 null bytes
+	programName := append([]byte("op-program"), 0x0)
```

Comment:
Should this also extend to hit the 4-byte alignment (for future-proofing)?
```suggestion
	programName := append([]byte("op-program"), 0x0, 0x0)
```

Reply:
Good idea.